core/arch/arm/include/kernel/early_ta.h

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2017, Linaro Limited
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice,
+ * this list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+#ifndef KERNEL_EARLY_TA_H
+#define KERNEL_EARLY_TA_H
+
+#include <compiler.h>
+#include <stdint.h>
+#include <tee_api_types.h>
+
+struct early_ta {
+	TEE_UUID uuid;
+	uint32_t size;
+	uint32_t uncompressed_size; /* 0: not compressed */
+	const uint8_t ta[]; /* @size bytes */
+} __aligned(8);
```

Comment:
At first I used `__alignof__(struct early_ta)` (instead of hard-coding 8) for rounding up the size of structs when iterating ([here](https://github.com/OP-TEE/optee_os/pull/1733/files#diff-019cb3df8fa51f3fa4eeda70130be846R48)). It was OK on QEMU but I was 4 bytes off on D02 (64-bits). So I forced 8-byte alignement as an easy fix ;)
I'll take a closer look and see if I can get rid of it.